### PR TITLE
Added Week 6 solution and guide

### LINKED ISSUE #4: New Year Chaos
[Problem Link](https://www.hackerrank.com/challenges/new-year-chaos)

### PR DESCRIPTION
**Pull request by asclines**
Closes #4 
Thoughts on format and file organization?

### CODE REVIEW

File: Guide/Constructive/New Year Chaos/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# New Year Chaos
+This problem comes from [HackerRank](https://www.hackerrank.com/challenges/new-year-chaos)
+and is categorized as a Constructive Algorithm
```
Comment (isaac-gs) [inline Guide/Constructive/New Year Chaos/README.md]: .?

File: Guide/Constructive/New Year Chaos/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# New Year Chaos
+This problem comes from [HackerRank](https://www.hackerrank.com/challenges/new-year-chaos)
+and is categorized as a Constructive Algorithm
+
+HackerRank Difficulty: Medium
+
+
+## Problem
+### Statement
```
Comment (isaac-gs) [inline Guide/Constructive/New Year Chaos/README.md]: Should we say that this is referenced from Hackerrank? I would actually suggest replacing this section with a link to the problem statement in Hackerrank or a short explanation by you. Their TOS actually says we can't use content from their site directly.

File: Guide/Constructive/New Year Chaos/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+That is to say, if _n=8_ and _Person 5_ bribes _Person 4_, the queue will look like this: _1,2,3,5,4,6,7,8_.
+
+Fascinated by this chaotic queue, you decide you must know the minimum number of bribes that took place to get the queue into its current state!
+
+Note: Each _Person X_ wears sticker _X_, meaning they were initially the _Xth_ person in queue.
+
+#### Input Format
+
+The first line contains an integer, _T_ , denoting the number of test cases.
```
Comment (isaac-gs) [inline Guide/Constructive/New Year Chaos/README.md]: Same issue as above.

File: Guide/Constructive/New Year Chaos/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+Note: Each _Person X_ wears sticker _X_, meaning they were initially the _Xth_ person in queue.
+
+#### Input Format
+
+The first line contains an integer, _T_ , denoting the number of test cases.  
+Each test case is comprised of two lines; the first line has _n_ (an integer indicating the number of people in the queue), and the second line has _n_ space-separated integers describing the final state of the queue.
+
+### Overview
+Here is a breakdown of what the problem is really asking.
```
Comment (isaac-gs) [inline Guide/Constructive/New Year Chaos/README.md:12]: I like this, I say replace the two above sections with just this.

File: Guide/Constructive/New Year Chaos/solution.cpp [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      /*
+         Note we are iterating backwards and arranging people as we go.
+         This means that at any point, we can assume everybody behind that point
+            is where they are supposed to be.
+          This forces the situation where if
+            line[i] != i, then either line[i-1] = i or line[i-2] = i
+       */
+      for (n = N; n > 0; n--) {
+        /*
+           We only care about 3 people & 3 positions at a time.
```
Comment (isaac-gs) [inline Guide/Constructive/New Year Chaos/solution.cpp]: Should we consider it standard to have this much documentation? Clean Code states that it's better to have code that is so easy to read that it doesn't need documentation. Although I would say that we should have greater than average documentation.

**Review by isaac-gs — DISMISSED** [review]
Overall looks good.

File: Guide/Constructive/New Year Chaos/README.md (hunk 1 above)
Comment (asclines) [reply Guide/Constructive/New Year Chaos/README.md]: What?

File: Guide/Constructive/New Year Chaos/README.md (hunk 2 above)
Comment (asclines) [reply Guide/Constructive/New Year Chaos/README.md]: Oh boy.. I was just going off of Orions suggestion for incase the problem gets taken down.  I'll remove it.

File: Guide/Constructive/New Year Chaos/solution.cpp (hunk 5 above)
Comment (asclines) [reply Guide/Constructive/New Year Chaos/solution.cpp]: Eh no.  I should take this out.  The comments predate the Readme and are from when this file was a standalone. I'll leave comments that don't get covered by the readme

File: Guide/Constructive/New Year Chaos/README.md (hunk 2 above)
Comment (asclines) [reply Guide/Constructive/New Year Chaos/README.md]: Also this is a link to the problem statement..

File: Guide/Constructive/New Year Chaos/README.md (hunk 1 above)
Comment (isaac-gs) [reply Guide/Constructive/New Year Chaos/README.md]: You're missing a . after your sentence.

File: Guide/Constructive/New Year Chaos/README.md (hunk 2 above)
Comment (isaac-gs) [reply Guide/Constructive/New Year Chaos/README.md]: It's a valid point, but I don't think there's a huge risk of them removing the content. Yeah, I guess I meant only have the problem statement along with a sentence that tells someone to go there.

File: Guide/Constructive/New Year Chaos/solution.cpp (hunk 5 above)
Comment (isaac-gs) [reply Guide/Constructive/New Year Chaos/solution.cpp]: Sounds good

**Review by isaac-gs — APPROVED** [review]

**Pull request merged**